registry/storage/catalog.go

```diff
@@ -95,3 +95,23 @@ func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error)
 	return nil
 
 }
+
+func pathGreaterThan(pathX, pathY string) (b bool) {
+	splitPathX := strings.SplitN(pathX, "/", 2)
+	splitPathY := strings.SplitN(pathY, "/", 2)
+
+	if splitPathX[0] == splitPathY[0] {
+		if len(splitPathX) == 1 && len(splitPathY) == 1 {
+			return false
+		} else if len(splitPathX) == 1 && len(splitPathY) != 1 {
+			return false
+		} else if len(splitPathX) != 1 && len(splitPathY) == 1 {
+			return true
+		}
+
+		return pathGreaterThan(splitPathX[1], splitPathY[1])
```

Comment:
You have to pass in the rest of the path here (rather than just the next element) because repositories can have more than two levels.

e.g. this would not work  
```
foo-bar/b
foo-bar/baz-bat/a
foo-bar/baz-bat/b
```